mutagen/mp4/__init__.py

```diff
@@ -891,6 +892,119 @@ def to_line(key, value):
         return u"\n".join(values)
 
 
+class MP4Chapters(object):
+    """MP4Chapters()
+
+    MPEG-4 Chapter information.
+
+    Supports the 'moov.udta.chpl' box.
+
+    Behaves like a list of tuples where each tuple contains:
+        timestamp (`int`): position from the start of the file in seconds
+        title (`str`): title of the chapter
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._timescale = None
+        self._duration = None
+        self._chapters = []
+        super(MP4Chapters, self).__init__()
+        if args or kwargs:
+            self.load(*args, **kwargs)
+
+    def __iter__(self):
+        iter(self._chapters)
+
+    def __delitem__(self, key):
+        self._chapters.__delattr__(key)
+
+    def __getitem__(self, key):
+        return self._chapters.__getattribute__(key)
+
+    def __setitem__(self, key, value):
+        self._chapters.__setattr__(key, value)
+
+    def load(self, atoms, fileobj):
+        try:
+            mvhd = atoms.path(b"moov", b"mvhd")[-1]
+        except KeyError as key:
+            return MP4MetadataError(key)
+
+        self._parse_mvhd(mvhd, fileobj)
+
+        if not self._timescale:
+            raise MP4MetadataError("Unable to get timescale")
+
+        try:
+            chpl = atoms.path(b"moov", b"udta", b"chpl")[-1]
+        except KeyError as key:
+            return MP4MetadataError(key)
+
+        self._parse_chpl(chpl, fileobj)
+
+    @classmethod
+    def _can_load(cls, atoms):
+        return b"moov.udta.chpl" in atoms
```

Comment:
Shouldn't this check for moov.mvhd as well?

Reply:
Implemented.